lib/issuer.js

```diff
@@ -65,8 +65,12 @@ exports.issue = async ({credential, profileAgentRecord}) => {
     } catch(e) {
       if(!(e.name === 'InvalidStateError' && credentialStatusWriter &&
         await credentialStatusWriter.exists({credential}))) {
-        // error not caused by a duplicate credential status, throw it
-        throw e;
+        throw new BedrockError(
+          'Duplicate credential id.',
+          'DuplicateError', {
+            public: true,
+            httpStatusCode: 409
+          }, e);
```

Comment:
This doesn't seem quite right. I think this error would be thrown for any underlying database error which would not correctly reflect whatever that underlying problem was. I think we need to try and do a `countDocuments` operation on the credentials collection using the credential's ID and then only if we find a result should we throw this special error. Otherwise, we should just `throw e;`. So something like:

```js
// error not caused by a duplicate credential status ID and needs to be thrown;
// first see if it was caused by a duplicate credential ID and throw a specific
// error -- otherwise throw whatever the error is
if(await credentialsCollection.countDocuments({
  equals: {'content.credentialStatus.id': credential.credentialStatus.id}
}) !== 0) {
  throw new BedrockError(
    'Could not issue credential; duplicate credential ID.',
    'DuplicateError', {
      public: true,
      httpStatusCode: 409
    });
}
throw e;
```